docs/future_work/NewArchitecture.md

```diff
@@ -0,0 +1,200 @@
+
+
+## Detailed Description
+
+In this section we will provide a detailed description of the steps necessary for the architecture to function. In particular we will be discussing:
+
+  * Uploading Data
+  * Performing a User Requested Algorithm
+  * Participating in an Aggreagated Algorithm
+
+This section is important because the steps will be very close to the actual implementation. If there is an issue that is clear from this section it suggests the new for at a minimum an implementation redesign and possibly architectural changes.
+
+### Uploading Data
+
+The following are the steps that are necessary to upload data to the server.
+
+1. The user's smartphone makes a request to a known access location (essential a server at a known domain) with a request to spawn a user cloud instance.
+2. The known access location spawns a container to produce a "user cloud." This user cloud consists of a server running inside a secure enclave via Graphene. The known access location then replies to the smartphone with an address and port of the spawned user cloud.
+3. The smart phone connects to the known access location. The two establish a secure channel through SGX's remote attestation. All user cloud will run the same general program, so this component is trusted to only allow a new user to connect once at the beginning. While the known access location is untrusted the user cloud code's will be open source and its hash known, allowing us to verify the connection. Then the smartphone will send its private key and profile of allowed algorithm to the user cloud.
+4. The user sends some data to the user cloud that it wishes to store over the established secure connection.
+5. The user cloud spawns the user's database instance as a container and provides the instance with the private key. The instance can be any paricular database which runs on a section of a distributed file system reserved just for the user (so all contents can be encrypted with the user's private key).
+6. The user cloud sends the data to the database instance. This database instance will then store the data encrypted with the private key.
+
+Steps 1-3 constitute the process of launching a user cloud. If the user cloud is already running then in step 2 rather than launch a new user cloud the known access location should just return the address of the user's user cloud which is already running (which it should be possible to authenticate, although we may want to produce some shared secret for existing user clouds).
```

Comment:
do we need to authenticate? Since the known access location is untrusted, it can return an arbitrary user cloud. The phone should attest the user cloud before it communicates with it, presumably through the cert mechanism.